compiler/rustc_middle/src/mir/mod.rs

```diff
@@ -1858,41 +1858,55 @@ pub struct CopyNonOverlapping<'tcx> {
 
 /// Places roughly correspond to a "location in memory." Places in MIR are the same mathematical
 /// object as places in Rust. This of course means that what exactly they are is undecided and part
-/// of the Rust memory model. However, they will likely contain at least the following three pieces
-/// of information in some form:
+/// of the Rust memory model. However, they will likely contain at least the following pieces of
+/// information in some form:
 ///
-///  1. The part of memory that is referred to (see discussion below for details).
-///  2. The type of the place and an optional variant index. See [`PlaceTy`][tcx::PlaceTy]
-///  3. The provenance with which the place is being accessed.
+///  1. The address in memory that the place refers to.
+///  2. The provenance with which the place is being accessed.
+///  3. The type of the place and an optional variant index. See [`PlaceTy`][tcx::PlaceTy].
+///  4. Optionally, some metadata. This exists if and only if the type of the place is not `Sized`.
 ///
-/// We'll give a description below of how the first two of these three properties are computed for a
-/// place. We cannot give a description of the provenance, because that is part of the undecided
-/// aliasing model - we only include it here at all to acknowledge its existence.
+/// We'll give a description below of how all pieces of the place except for the provenance are
+/// calculated. We cannot give a description of the provenance, because that is part of the
+/// undecided aliasing model - we only include it here at all to acknowledge its existence.
 ///
-/// For a place that has no projections, ie `Place { local, projection: [] }`, the part of memory is
-/// the local's full allocation and the type is the type of the local. For any other place, we
-/// define the values as a function of the parent place, that is the place with its last
-/// [`ProjectionElem`] stripped. The way this is computed of course depends on the kind of that last
-/// projection element:
+/// Each local naturally corresponds to the place `Place { local, projection: [] }`. This place has
+/// the address of the local's allocation and the type of the local.
+///
+/// **Needs clarification:** Unsized locals seem to present a bit of an issue. Their allocation
+/// can't actually be created on `StorageLive`, because it's unclear how big to make the allocation.
+/// Furthermore, MIR produces assignments to unsized locals, although that is not permitted under
+/// `#![feature(unsized_locals)]` in Rust. Besides just putting "unsized locals are special and
+/// different" in a bunch of places, I (JakobDegen) don't know how to incorporate this behavior into
+/// the current MIR semantics in a clean way - possibly this needs some design work first.
```

Comment:
Yeah. We have a hack in Miri that I can explain to you, but I would love for someone to come up with a cleaner solution. :)